src/deepcode/lib/modules/LoginModule.ts

```diff
@@ -23,31 +26,38 @@ abstract class LoginModule extends ReportModule implements DeepCode.LoginModuleI
     try {
       const checkCurrentToken = await this.checkSession();
       if (checkCurrentToken) return;
-      const result = await http.login(this.baseURL, this.source);
-      const { sessionToken, loginURL } = result;
-      if (!sessionToken || !loginURL) {
+      const response = await startSession({ baseURL: this.baseURL, source: this.source });
+      if (response.type == 'error') {
         throw new Error(errorsLogs.login);
       }
-      await this.setToken(sessionToken);
+
+      const { sessionToken, loginURL } = response.value;
+
       await viewInBrowser(loginURL);
-      await this.waitLoginConfirmation();
+      const confirmed = await this.waitLoginConfirmation(sessionToken);
+      if (confirmed) {
+        await this.setToken(sessionToken);
+      }
     } catch (err) {
       await this.processError(err, {
-        message: errorsLogs.login
+        message: errorsLogs.login,
       });
     } finally {
       this.pendingLogin = false;
     }
   }
 
-  async checkSession(): Promise<boolean> {
+  async checkSession(token = ''): Promise<boolean> {
     let validSession = false;
-    if (this.token) {
+    if (token || this.token) {
       try {
-        validSession = !!(await http.checkSession(this.baseURL, this.token));
+        validSession = !!(await checkSession({
```

Comment:
This also doesn't work anymore with the current tsc response (it returns true even for 304 errors). It should be something like:

`const session = await checkSession({
          baseURL: this.baseURL,
          sessionToken: token || this.token,
        });
        if (session.type === 'error') {
          validSession = false;
        } else {
          validSession = session.value;
        }`

Reply:
done